### PR TITLE
Revert propagation of drop-live information from Polonius

### PR DESCRIPTION
**Pull request by amandasystems**
#64749 introduced a flow of drop-use data from Polonius to `LivenessResults::add_extra_drop_facts()`, which makes `LivenessResults` agree with Polonius on liveness in the presence of free regions that may be dropped. Later changes accidentally removed this flow. This PR restores it.

### CODE REVIEW

**Comment by amandasystems** [comment]
Ok, apparently I did this on the wrong branch. Fixing it now!

**Comment by amandasystems** [comment]
Done!

**Comment by oli-obk** [comment]
cc @matthewjasper

**Comment by amandasystems** [comment]
Note the FIXME comment: I tried to figure out a way to neatly pass the fact data (which has finished baking and is safe to share) with `LivenessResults` but couldn't come up with anything more elegant.

**Comment by oli-obk** [comment]
Yea I like the new version!

@bors r+ rollup

File: compiler/rustc_borrowck/src/type_check/liveness/trace.rs [hunk 1]
```diff
@@ -218,21 +216,38 @@ impl<'me, 'typeck, 'flow, 'tcx> LivenessResults<'me, 'typeck, 'flow, 'tcx> {
...
-        relevant_live_locals: FxIndexSet<Local>,
-    ) {
+    fn add_extra_drop_facts(&mut self, relevant_live_locals: &[Local]) -> Option<()> {
+        let drop_used = self
+            .cx
+            .typeck
+            .borrowck_context
+            .all_facts
+            .as_ref()
+            .map(|facts| facts.var_dropped_at.clone())?;
```
Comment (lqd) [inline compiler/rustc_borrowck/src/type_check/liveness/trace.rs:226]: drive-by question: how big is this vec?

Comment (amandasystems) [reply compiler/rustc_borrowck/src/type_check/liveness/trace.rs:226]: When I wrote my master’s thesis drop-uses weren’t super common so I’d say small ish, but I’d like to remove this clone if possible. 

Of course I had a dog walk idea how to do that just in time for it to get merged before I’d implemented it but I’ll look into it next time I need something to pass the time

Comment (amandasystems) [reply compiler/rustc_borrowck/src/type_check/liveness/trace.rs:226]: I should add that the clone happened in the original version too, except not as a clone but as a for each insert.
